src/crud/message.py

```
@@ -237,7 +238,57 @@ async def create_messages(
+        _existing_rows = await db.execute(
+            select(models.Message.peer_name, models.Message.content).where(
+                models.Message.workspace_name == workspace_name,
+                models.Message.session_name == session_name,
+                models.Message.created_at >= _lookback,
+            )
+        )
+        for _row in _existing_rows.scalars().all():
+            # Composite key: peer_name + content prevents false positives
+            # when different peers say the same thing (e.g. "ok", "thanks")
+            _composite = f"{_row[0]}|{_row[1]}" if isinstance(_row, tuple) else str(_row)
+            _existing_hashes.add(md5(_composite.encode()).hexdigest())
```

Ruff (0.15.12)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)